client/pov-recovery/src/lib.rs

```diff
@@ -597,16 +606,27 @@ where
 }
 
 /// Returns a stream over pending candidates for the parachain corresponding to `para_id`.
-async fn pending_candidates(
+async fn pending_candidates<Block: BlockT>(
 	relay_chain_client: impl RelayChainInterface + Clone,
 	para_id: ParaId,
+	sync_service: Arc<SyncingService<Block>>,
 ) -> RelayChainResult<impl Stream<Item = (CommittedCandidateReceipt, SessionIndex)>> {
 	let import_notification_stream = relay_chain_client.import_notification_stream().await?;
 
 	let filtered_stream = import_notification_stream.filter_map(move |n| {
 		let client_for_closure = relay_chain_client.clone();
+		let sync_oracle = sync_service.clone();
 		async move {
 			let hash = n.hash();
+			if sync_oracle.is_major_syncing() {
```

Comment:
Is it going to download the candidate later if it now skips them since it's syncing or what is going to happen with this pending candidate?

Reply:
We are maintaining a list of pending candidates that we have seen from the relay chain but that have not yet been finalized. So the list of candidates we maintain here is limited anyway. During sync, we skip them now, but once we reach the tip, the list will soon be complete again.